src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: registering operator has less than kickPercentageOfOperatorStake"
+                );
+                
+                // check that the operator to kick has less than the kick percentage of the average stake
+                require(
+                    operatorToKickStake < totalStakeForQuorum * operatorSetParam.kickPercentageOfAverageStake / 100 / numOperatorsForQuorum,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: operator to kick has more than kickPercentageOfAverageStake"
+                );
+                // check the that the operator to kick has lss than the kick percentage of the total stake
+                require(
+                    operatorToKickStake < totalStakeForQuorum * operatorSetParam.kickPercentageOfTotalStake / 100,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: operator to kick has more than kickPercentageOfTotalStake"
+                );
+            }
+            
+            // kick the operator
+            _deregisterOperatorWithCoordinator(
+                operatorKickParams[i].operator, 
+                quorumNumbers[i:i+1], 
+                operatorKickParams[i].pubkey, 
+                operatorKickParams[i].operatorIdsToSwap, 
+                operatorKickParams[i].globalOperatorListIndex
+            );
+        }
+    }
+
+    /**
+     * @notice Deregisters the msg.sender as an operator from the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registered for
+     * @param deregistrationData is the the data that is decoded to get the operator's deregisteration information
+     * @dev `deregistrationData` should be a tuple of the operator's BLS public key, the list of operator ids to swap, 
+     * and the operator's index in the global operator list
+     */
+    function deregisterOperatorWithCoordinator(bytes calldata quorumNumbers, bytes calldata deregistrationData) external {
+        // get the operator's deregisteration information
+        (BN254.G1Point memory pubkey, bytes32[] memory operatorIdsToSwap, uint32 globalOperatorListIndex) 
+            = abi.decode(deregistrationData, (BN254.G1Point, bytes32[], uint32));
+        // call internal function to deregister the operator
+        _deregisterOperatorWithCoordinator(msg.sender, quorumNumbers, pubkey, operatorIdsToSwap, globalOperatorListIndex);
+    }
+
+    /**
+     * @notice Deregisters the msg.sender as an operator from the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registered for
+     * @param pubkey is the BLS public key of the operator
+     * @param operatorIdsToSwap is the list of the operator ids that the should swap for the deregistering operator's index
```

Comment:
some typo here. `that the should swap for` needs to change, I think.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/b03c92cfbb9bfd3366ad0bb1c11c20d3cedf29c1